### DISCUSSION TITLE
Database.beginGlobalTransaction as common practice

### DISCUSSION THREAD

**Discussion by yodosan** [adonisjs/core#2274] [Help] [answered]
Hello,
I was wondering if it possible to use the "global transaction" approach more extensively, like in this example:

```
import Database from '@ioc:Adonis/Lucid/Database'

await Database.beginGlobalTransaction()
try {
  await UserService.store({ ... })
  await RoleService.store({ ... })
  await Database.commitGlobalTransaction()
} catch (e) {
  Logger.error(e)
  await Database.rollbackGlobalTransaction()
}
```

or there are some drawbacks. I think there are, because I found this discussion [here](https://github.com/adonisjs/core/discussions/1912) and the statement "It will effect parallel queries only if they are part of the same process" makes me think that this above is not a good approach, but I would have a confirmation about it.

Thank you!

**Comment by thetutlage** [comment, marked as answer]
Yes, you should not use it in actual code. It is mainly for tests
